File: src/main/kotlin/mocking/tpp/models/NationalId.kt
package mocking.tpp.models

import javax.xml.bind.annotation.XmlAccessType
import javax.xml.bind.annotation.XmlAccessorType
import javax.xml.bind.annotation.XmlAttribute
import javax.xml.bind.annotation.XmlValue

@XmlAccessorType(XmlAccessType.FIELD)
data class NationalId(
        @XmlAttribute var type: String = "default type",
        @XmlValue var value: String = "default value"
)

File: backendworker/NHSOnline.Backend.GpSystems/Suppliers/Tpp/Models/NationalId.cs
using System.Xml.Serialization;

namespace NHSOnline.Backend.GpSystems.Suppliers.Tpp.Models
{
    public class NationalId
    {
        [XmlAttribute("type")]
        public string Type { get; set; }

        [XmlText]
        public string Value { get; set; }
    }
}